metaflow/client/core.py

```diff
@@ -303,7 +303,7 @@ def __init__(
             # distinguish between "attempt will happen" and "no such
             # attempt exists".
 
-        if pathspec:
```

Comment:
Yes -- we discussed this and this seems like a bug. The code makes no snese to check just for pathspec beause the else also sets self._pathspec so we can clearly have both (and there is a valid use to have both).